decryptor/postgresql/protocol.go

```diff
@@ -151,19 +154,34 @@ func (p *PgProtocolState) HandleDatabasePacket(packet *PacketHandler) error {
 	}
 
 	if packet.IsParseComplete() {
+		p.gotParseComplete = true
 		p.lastPacketType = ParseCompletePacket
 		return nil
 	}
 
 	if packet.IsBindComplete() {
+		p.gotBindComplete = true
 		p.lastPacketType = BindCompletePacket
 		return nil
 	}
 
 	// ReadyForQuery starts a new query processing. Forget pending queries.
 	// There is nothing interesting in the packet otherwise.
 	if packet.IsReadyForQuery() {
-		p.forgetQueryState()
+		// Forget pending parse, but only if we got ParseComplete before
+		if p.gotParseComplete {
+			p.forgetPendingParse()
+			p.gotParseComplete = false
+		}
+
+		// Forget pending bind, but only if we got BindComplete before
+		if p.gotBindComplete {
+			p.forgetPendingBind()
+			p.gotBindComplete = false
+		}
+
+		p.forgetPendingExecute()
+		p.forgetPendingQuery()
```

Comment:
should we forget query after ParseComplete (actually packet that stores query)? Or any next packets depends on that (Bind/Execute)?